pkg/observer/observer.go

```diff
@@ -341,8 +341,8 @@ func (k *Observer) Start(ctx context.Context) error {
 }
 
 // InitSensorManager starts the sensor controller
-func (k *Observer) InitSensorManager() error {
-	mgr, err := sensors.StartSensorManager(option.Config.BpfDir)
+func (k *Observer) InitSensorManager(keepSensorsOnExit bool) error {
+	mgr, err := sensors.StartSensorManager(option.Config.BpfDir, keepSensorsOnExit)
```

Comment:
This is not an objection, but I would like for us to have a consistent way of accessing the config options. 

Some time ago @jrfastab did https://github.com/cilium/tetragon/commit/64b236be12e6feb440fbc718ac1f8460bd8e674a, where the idea (as I'm reading it) was to add configuration to under `option.Config` and use that from everywhere.

The above change goes into a different direction and passes `option.Config.KeepSensorsOnExit`. I don't have a strong opinion on what approach should we take, but I do think we should be consistent.

If we want to avoid pass the argument and use `option.Config.KeepSensorsOnExit` one complication is that it makes it tricky to do testing because the default config value might change and you might want to test a specific value.

If that's the issue here, one of the ways to deal with it is to save the default value, set the value you want and restore the default value after the test. See: https://github.com/cilium/tetragon/blob/93939d149633c648bc5cc17fd157145a2d034e4a/pkg/policyfilter/k8s_test.go#L712-L719

Reply:
the issue is that we check option.Config.KeepSensorsOnExit un Unload but we don't know if we unload because tetragon is going down or we unload because we just removed policy.. both these cases have different path:
- when tetragon is going down we want to follow  KeepSensorsOnExit
- when we just unload sensor we DO NOT want to follow KeepSensorsOnExit

so that's why I store that in sensor manager and use it in tetragon exit path for sensor unload
I should have put this in some comments, sorry

Reply:
ok, using KeepSensorsOnExit directly from SensorManager::RemoveAllSensors function, thanks